### PR TITLE
fix: cleanup signal listener

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

**Review by mcollina — COMMENTED** [review]
Can you please add a unit test?

File: lib/fetch/request.js [hunk 1]
```diff
@@ -328,14 +332,9 @@ class Request {
...
-        // TODO: Remove this listener on failure/success.
-        signal.addEventListener(
-          'abort',
-          function () {
-            ac.abort()
-          },
-          { once: true }
-        )
+        const abort = () => ac.abort()
+        signal.addEventListener( 'abort', abort, { once: true })
```
Comment (ronag) [inline lib/fetch/request.js:336]:
```suggestion
        signal.addEventListener('abort', abort, { once: true })
```

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by ronag** [comment]
@benjamingr @dominic not the prettiest solution... any suggestions? The spec doesn't specify how and when the abort listener should be cleaned up.

**Comment by benjamingr** [comment]
Remove the abort listener when you know you are not going to abort (when the request finishes, was aborted, errored etc).

FinalizationRegistry is a really heavy handed tool for this

File: lib/fetch/request.js [hunk 2]
```diff
@@ -328,14 +332,9 @@ class Request {
...
-        signal.addEventListener(
-          'abort',
-          function () {
-            ac.abort()
-          },
-          { once: true }
-        )
+        const abort = () => ac.abort()
+        signal.addEventListener('abort', abort, { once: true })
+        requestFinalizer.register(this, { signal, abort })
```
Comment (mcollina) [inline lib/fetch/request.js:337]: We should be unregistering this after successful completion. Always passing through the registry is expensive.

**Comment by ronag** [comment]
> Remove the abort listener when you know you are not going to abort (when the request finishes, was aborted, errored etc).

It's a little more complicated 😄. The spec doesn't have any request finalisation so I would have to make up something myself.

**Comment by benjamingr** [comment]
Oh that's a DOM fetch request? Yeah, I think cheating is probably fine the browser cheats by doing this internally

File: lib/fetch/request.js (hunk 2 above)
Comment (ronag) [reply lib/fetch/request.js:337]: Yea, but if we could do that we wouldn't need this hack at all.

**Comment by ronag** [comment]
@mcollina PTAL 82e1a5180a8b850c5d1c11a6653278d792241584

**Pull request merged**